pandas/tests/indexing/test_loc.py

```diff
@@ -628,3 +628,22 @@ def test_loc_empty_list_indexer_is_ok(self):
         tm.assert_frame_equal(df.loc[[]], df.iloc[:0, :],
                               check_index_type=True,
                               check_column_type=True)
+
+
+def test_loc_datetime_length_one():
+    """
+    Test if DataFrame with length one is properly sliced
+    by .loc.
+    """
+    from datetime import datetime
+    df = pd.DataFrame(columns=['1'],
+                      index=pd.date_range('2016-10-01T00:00:00',
+                                          '2016-10-01T23:59:59'))
+    df_sliced_dt = df.loc[datetime(2016, 10, 1):]
+    tm.assert_frame_equal(df, df_sliced_dt)
+
+    df = pd.DataFrame(columns=['1'],
```

Comment:
This is the same `df` right? Shouldn't need to re-create it (unless I'm missing something).